nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestAnnotationNotification.java

```diff
@@ -153,23 +159,52 @@ protected void captureAndVerifyAnnotationEventNotification(Supplier<DocumentMode
             assertEquals(1, handledEvents.size());
             Event expectedEvent = handledEvents.get(0);
             assertEquals(annotationEventType, expectedEvent.getName());
-            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotatedDocumentModel);
-            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel,
-                    handledEvents.get(0), annotationEventType);
+
+            checkDocumentEventContext(expectedEvent, annotationDocumentModel, annotationParentDocumentModel,
+                    annotatedDocumentModel);
+
+            checkReceivedMail(emailsResult.getMails(), annotationDocumentModel, annotatedDocumentModel, handledEvents.get(0),
+                    annotationEventType);
+            return annotationDocumentModel;
         }
     }
 
+    @Test
+    public void shouldNotifyWithTheRightAnnotatedDocument() {
+        // First comment
+        Annotation createdAnnotation = createAnnotation(annotatedDocumentModel);
+        DocumentModel createdAnnotationDocModel = session.getDocument(new IdRef(createdAnnotation.getId()));
+        // before subscribing, or previous event will be notified as well
+        transactionalFeature.nextTransaction();
+        // Reply
+        captureAndVerifyAnnotationEventNotification(() -> {
+            // subscribe to notifications
+            NuxeoPrincipal principal = session.getPrincipal();
+            String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
+            String notif = "CommentAdded";
+            notificationService.addSubscription(subscriber, notif, annotatedDocumentModel, false, principal, notif);
```

Comment:
i think we can extract this into method that will be used on `createAnnotationAndAddSubscription..`
```
protected void addSubscriptions(String... notifications) {
        NuxeoPrincipal principal = session.getPrincipal();
        String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
        for (String notif : notifications) {
            notificationService.addSubscription(subscriber, notif, annotatedDocumentModel, false, principal, notif);
        }
    }
```

```
 protected Annotation createAnnotationAndAddSubscription(String... notifications) {
        addSubscriptions(notifications);

        return createAnnotation(annotatedDocumentModel);
    }
```

and on your code 
```
  captureAndVerifyAnnotationEventNotification(() -> {
            // subscribe to notifications
            addSubscriptions("CommentAdded");

            Comment reply = createAnnotation(createdAnnotationDocModel);
            DocumentModel replyDocumentModel = session.getDocu
....
```

Reply:
Yes done !